ui/src/message_popup/experience_page.jsx

```diff
@@ -46,42 +42,19 @@ export default React.createClass({
 
   getInitialState: function() {
     const isSolutionMode = _.has(this.props.query, 'solution');
-    const helpType = isSolutionMode ? 'none' : 'feedback';
-    const shouldShowStudentCard = isSolutionMode ? _.has(this.props.query, 'cards') : true;
-    const shouldShowSummary = !isSolutionMode;
-    const sessionId = uuid.v4();
     return {
-      shouldShowStudentCard,
-      shouldShowSummary,
-      helpType,
-      isSolutionMode,
-      sessionId,
-      competencyGroupValue: ALL_COMPETENCY_GROUPS,
-      questions: allQuestions,
-      email: this.context.auth.userProfile.email,
-      hasStarted: false,
-      questionsAnswered: 0,
-      sessionLength: 10,
+      scaffolding: {
+        helpType: isSolutionMode ? 'none' : 'feedback',
+        shouldShowStudentCard: isSolutionMode ? _.has(this.props.query, 'cards') : true,
+        shouldShowSummary: !isSolutionMode,
+      },
       toastRevision: false,
```

Comment:
It'd be good to add in `gameSession: null`, to just to indicate that's part of the state but isn't set yet when the component first mounts.